### DISCUSSION TITLE
Question: How can I determine if hydrogen is connected to a non-carbon atom in the molecule?

### DISCUSSION THREAD

**Discussion by mohammad-saber** [rdkit/rdkit#4052] [Q&A] [answered]
Hi, thank you very much for reading my question. 
I am looking for a way to determine if any non-carbon atom in the molecule has Hydrogen connected to it. Example:

For smile `C(C)N` (shown below), Nitrogen is connected to one Carbon and two Hydrogen atoms. 

<img width="201" alt="Screen Shot 2021-04-19 at 13 47 07" src="https://user-images.githubusercontent.com/38423900/115182921-c9629b80-a115-11eb-84ee-8c9532ea0faf.png">

I am using the following code. 

```
smile = "C(C)N"
mol = Chem.MolFromSmiles(smile)
for atom in mol.GetAtoms():
    if atom.GetAtomicNum() != 6:   # if atom is not Carbon
        print("Non-carbon atom : ", atom.GetSymbol())
        print([x.GetSymbol() for x in atom.GetNeighbors()], "\n")

```

However, the above code only returns "C" as a neighbor to "N". So, I cannot determine if N is connected to H or not.  

Is there any way that I can determine "N" is connected to H atom?

Thank you very much for your help.

**Comment by greglandrum** [comment, marked as answer]
You probably want to use this method: http://www.rdkit.org/docs/source/rdkit.Chem.rdchem.html#rdkit.Chem.rdchem.Atom.GetTotalNumHs
